Sources/SourceKitBazelBSP/Server/MessageHandler/BSPMessageHandler.swift

```diff
@@ -132,3 +134,24 @@ final class BSPMessageHandler: MessageHandler {
         return handler
     }
 }
+
+final class LockIsolated<Value>: @unchecked Sendable {
+    private var value: Value
+    private let lock = NSRecursiveLock()
```

Comment:
Let me do some thinking on this. The thing is that `OSAllocatedUnfairLock` is much faster than `NSRecursiveLock `, and recursive lock in general are a sign of wrong design on our end. We might want to move things around a bit more instead or go for Swift concurrency.